### PR TITLE
Api

### PR DESCRIPTION
**Pull request by codehass**
# Add API endpoints
Hi 🖐, CR. In this pull request @codehass and @Wahaj-Ali we:
- Create an API endpoint to list all posts for a user.
- Create an API endpoint to list all comments for a user's post.
- Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should not be set by passing a parameter.
- Your API endpoints should receive JSON and respond to JSON as well.

#### API Endpoint :
   -  http://localhost:3000/users/:id/posts.json
   -  http://localhost:3000/users/:id/posts/post_id/comments.json
   
## Screenshots
api2
api1

Thank you for reviewing this pull request 😎😎

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -9,8 +16,24 @@ def create
...
+      format.html do
+        redirect_to user_post_path(@post.author, @post)
+      end
+      format.json do
+        render json: @comment
+      end
+    else
+      format.html do
+        { render: new }
+      end
```
Comment (Qoosim) [inline app/controllers/comments_controller.rb]: - [x] You're also required to create an API endpoint to add a comment to a post. This has not been implemented. Kindly attend to it.

### Project Requirement
add-comment

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Git Flow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (Wahaj-Ali) [reply app/controllers/comments_controller.rb]: Thank you @Qoosim for pointing out the short coming. I worked it out after your suggestion.

**Review by sumairq — APPROVED** [review]
### Hi @codehass , 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**